code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -337,3 +215,172 @@
 		M.adjustOxyLoss(1, 0)
 	..()
 	. = 1
+
+
+//Amphetamines
+//Amphetamine sulfate
+/datum/reagent/drug/amphetamine
+	name = "Amphetamine"
+	id = "amphetamine"
+	description = "A potent CNS stimulant renowned for it's innocuous metabolization. Dangerous in high doses."
+	reagent_state = LIQUID
+	color = "#FAFAFA"
+	overdose_threshold = 30
+	addiction_threshold = 20
+	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+
+
+/datum/reagent/drug/amphetamine/on_mob_life(mob/living/M)
+	var/high_message = pick("You feel hyper.", "You have a pit in your stomach.", "Your head is pounding.")
+	if(prob(3))
+		to_chat(M, "<span class='notice'>[high_message]</span>")
+	M.AdjustStun(-2, 0)
+	M.AdjustKnockdown(-2, 0)
+	M.AdjustUnconscious(-40, 0)
+	M.adjustStaminaLoss(-2, 0)
+	M.status_flags |= GOTTAGOFAST
+	M.Jitter(2)
+	if(prob(5))
+		M.emote(pick("twitch", "shiver"))
+	..()
+	. = 1
+
+/datum/reagent/drug/amphetamine/overdose_process(mob/living/M)
+	if(M.canmove && !ismovableatom(M.loc))
+		for(var/i in 1 to 4)
+			step(M, pick(GLOB.cardinals))
+	if(prob(33))
+		M.visible_message("<span class='danger'>[M] shivers violently!</span>")
+		var/obj/item/I = M.get_active_held_item()
+		if(I)
+			M.drop_item()
+	..()
+	. = 1
+
+
+/datum/reagent/drug/amphetamine/addiction_act_stage4(mob/living/M)
+	if(prob(30))
+		M.adjustStaminaLoss(2, 0)
+		to_chat(M, "<span class='boldannounce'>You're not feeling good at all! You really need some [name].</span>")
+	return
+
+
+
+//Methamphetamine
+/datum/reagent/drug/amphetamine/methamphetamine
+	name = "Methmphetamine"
+	id = "methamphetamine"
+	description = "Amphetamine with an extra methyl group attached to the N isonomer. While more potent, it is also much more toxic and introduces muscular atrophy due to cortisol release."
+	reagent_state = LIQUID
+	color = "#FAFAFA"
+	overdose_threshold = 20
+	addiction_threshold = 3
+	metabolization_rate = 1 * REAGENTS_METABOLISM //meth is metabolized quicker. usually because it's smoked or injected, but reagent metabolization code is super primitive so this will do.
+	var/heart_attack_chance = 0.001
+
+/datum/reagent/drug/amphetamine/methamphetamine/on_mob_life(mob/living/M)
+	var/high_message = pick("You feel hyper.", "You feel like you need to go faster.", "You feel like you can run the world.")
+	if(prob(5))
+		to_chat(M, "<span class='notice'>[high_message]</span>")
+	M.AdjustStun(-20, 0)
+	M.AdjustKnockdown(-20, 0)
+	M.AdjustUnconscious(-40, 0)
+	M.adjustStaminaLoss(-1, 0)//meth isn't actually that great of a stamina booster.
+	M.status_flags |= GOTTAGOREALLYFAST
+	M.Jitter(4)
+	M.adjustBrainLoss(0.25)
+	if(prob(5))
+		M.emote(pick("twitch", "shiver"))
+	if(prob(heart_attack_chance))
```

Comment:
use the heart attack disease we already have. i think you can start a disease at a higher stage than one if you want it to be faster.

Reply:
i feel like that'd be worse tbf, this is fairly clear and can be modified easier